packages/castmill/lib/castmill/addons/devices/components/device-view.tsx

```diff
@@ -1,17 +1,96 @@
-import { Component } from 'solid-js';
+import { Component, createSignal } from 'solid-js';
 import { Device } from '../interfaces/device.interface';
+import { Tabs, LoadingOverlay } from '@castmill/ui-common';
+import { Maintainance } from './maintainance';
+import { DeviceLogs } from './device-events';
+import { DeviceDetails, DeviceUpdate } from './device-details';
+import { DevicesService } from '../services/devices.service';
+import { DeviceCache } from './device-cache';
+
+// Optionally we should allow using protonmaps
+// https://protomaps.com/
 
 // Modal component that will be used to display the device details and allow the user to edit the device
-const DeviceView: Component<{ device: Device }> = ({ device }) => {
+const DeviceView: Component<{
+  device: Device;
+  organization_id: string;
+  onChange?: (device: Device) => void;
+}> = (props) => {
+  const [loading, setLoading] = createSignal(false);
+
+  const updateDevice = async (device: DeviceUpdate) => {
+    setLoading(true);
+
+    try {
+      await Promise.all([
+        await DevicesService.updateDevice(
+          props.organization_id,
+          props.device.id,
+          device
+        ),
+        new Promise((resolve) => setTimeout(resolve, 300)),
```

Comment:
What is the purpose of this promise? Do we always have to wait at least 300 ms? If so, a comment would be nice :)